deploy.py

```diff
@@ -117,47 +119,60 @@ def configure_resources(values, deploy_config):
     """Configure extra command line arguments for services."""
     for service_name, config in deploy_config["services"].items():
-        extra_cmd_args = []
-
-        for param in [
-            "max_batch_size",
-            "max_input_length",
-            "max_total_tokens",
-            "max_batch_total_tokens",
-            "max_batch_prefill_tokens",
-        ]:
-            if config.get(param):
-                extra_cmd_args.extend([f"--{param.replace('_', '-')}", str(config[param])])
-
-        if extra_cmd_args:
-            if service_name == "llm":
-                engine = config.get("engine", "tgi")
+        if service_name == "llm":
+            extra_cmd_args = []
+            engine = config.get("engine", "tgi")
+
+            # Define parameters based on engine type
+            if engine == "tgi":
+                params = [
+                    "max_batch_size",
+                    "max_input_length",
+                    "max_total_tokens",
+                    "max_batch_total_tokens",
+                    "max_batch_prefill_tokens",
+                ]
+            elif engine == "vllm":
+                params = [
+                    "max_num_seqs",
+                    "max_input_length",
+                    "max_total_tokens",
+                    "max_batch_total_tokens",
+                    "max_batch_prefill_tokens",
+                ]
+
+            for param in params:
+                param_value = config.get(param)
+                # Only add parameter if its value is not None and not empty string
+                if param_value is not None and param_value != "":
+                    extra_cmd_args.extend([f"--{param.replace('_', '-')}", str(param_value)])
+
+            if extra_cmd_args:
                 if engine not in values:
                     values[engine] = {}
                 values[engine]["extraCmdArgs"] = extra_cmd_args
-            else:
-                if service_name not in values:
-                    values[service_name] = {}
-                values[service_name]["extraCmdArgs"] = extra_cmd_args
 
     return values
 
 
 def configure_models(values, deploy_config):
     """Configure model settings for services."""
     for service_name, config in deploy_config["services"].items():
-        # Skip if no model_id defined or service is disabled
-        if not config.get("model_id") or config.get("enabled") is False:
+        # Get model_id and check if it's valid (not None or empty string)
+        model_id = config.get("model_id")
+        if not model_id or model_id == "" or config.get("enabled") is False:
             continue
 
         if service_name == "llm":
             # For LLM service, use its engine as the key
+            # Check if engine is valid (not None or empty string)
             engine = config.get("engine", "tgi")
-            values[engine]["LLM_MODEL_ID"] = config.get("model_id")
+            if engine and engine != "":
```

Comment:
Can engine be empty after that default of "tgi" in line 169?
If it can indeed be empty, then you need to set it to a default value so the model_id can be assigned correctly for the engine in line 171

Reply:
I think engine could not be empty.